src/connection.js

```diff
@@ -67,22 +71,22 @@ class Connection extends EventEmitter {
         throw new TypeError('The "config.authentication" property must be of type Object.');
       }
 
-      const type = config.authentication.type;
+      const authType = config.authentication.type;
       const options = config.authentication.options === undefined ? {} : config.authentication.options;
 
-      if (typeof type !== 'string') {
+      if (typeof authType !== 'string') {
         throw new TypeError('The "config.authentication.type" property must be of type string.');
       }
 
-      if (type !== 'default' && type !== 'ntlm' && type !== 'azure-active-directory-password' && type !== 'azure-active-directory-access-token') {
-        throw new TypeError('The "type" property must one of "default", "ntlm", "azure-active-directory-password" or "azure-active-directory-access-token".');
+      if (authType !== 'default' && authType !== 'ntlm' && authType !== 'azure-active-directory-password' && authType !== 'azure-active-directory-access-token' && authType !== 'azure-active-directory-MSI') {
```

Comment:
How do you feel about changing the authentication type to `azure-active-directory-msi` (all lower case) to match the naming of the other authentication types more closely?